app/javascript/controllers/autosave_status_controller.ts

```diff
@@ -0,0 +1,98 @@
+import invariant from 'tiny-invariant';
+import {
+  enable,
+  disable,
+  hasClass,
+  addClass,
+  removeClass,
+  ResponseError
+} from '@utils';
+import { z } from 'zod';
+
+import { ApplicationController } from './application_controller';
+
+const Gon = z.object({
+  autosave: z.object({ status_visible_duration: z.number() })
+});
+
+declare const window: Window & typeof globalThis & { gon: unknown };
+const { status_visible_duration } = Gon.parse(window.gon).autosave;
+
+const AUTOSAVE_STATUS_VISIBLE_DURATION = status_visible_duration;
+
+export class AutosaveStatusController extends ApplicationController {
+  connect(): void {
+    this.onGlobal('autosave:enqueue', () => this.onEnqueue());
+    this.onGlobal('autosave:end', () => this.onSucceed());
+    this.onGlobal<CustomEvent>('autosave:error', (event) =>
+      this.onError(event)
+    );
+  }
+
+  onRetry() {
```

Comment:
J'ai l'impression d'avoir du mal à faire la différence entre les évènements internes (clic sur retry) et externes (enqueue). P-ê une convention de nommage pour distinguer les deux ?

```suggestion
  didClickRetry() {
```